korge-fleks/src/commonMain/kotlin/com/github/quillraven/fleks/collection/bitArray.kt

```diff
@@ -140,10 +141,11 @@ class BitArray(
     }
 
     override fun equals(other: Any?): Boolean {
-        if (other == null) return false
         if (this === other) return true
+        if (other !is BitArray) throw FleksBitArrayTypeException(
```

Comment:
Equals shouldn't throw: https://docs.oracle.com/javase/7/docs/api/java/lang/Object.html#equals(java.lang.Object)

instead it should return false.
Suppose for example you have a set of mixed objects, using a HashSet and one object of a different type collides its hashCode. It the implementation will check object.equals() with objects with the same hashCode but different type. And it will expect to be false for different instances, and to not throw

Reply:
So `if (other !is BitArray) return false`

Reply:
You are right. I will change it :)